### ISSUE TITLE
Race in `kubeletcertrotate_test.go` around HTTP paths

### ISSUE THREAD

**Issue by s0j** [k0sproject/k0s#2340]
### Before creating an issue, make sure you've checked the following:

- [X] You are running the latest released version of k0s
- [X] Make sure you've searched for existing issues, both open and closed
- [X] Make sure you've searched for PRs too, a fix might've been merged already
- [X] You're looking at docs for the released version, "main" branch docs are usually ahead of released versions.

### Version

main

### What happened?

https://github.com/k0sproject/k0s/actions/runs/3363305811/jobs/5576317139

A recent PR run resulted in a `kubeletcertrotate_test.go` panic.

```
    kubeletcertrotate_test.go:90: waiting to see kubelet rotating the client cert before triggering Plan creation
    kubeletcertrotate_test.go:182: Applying autopilot plan #0
    kubeletcertrotate_test.go:141: kubectl apply output: 'plan.autopilot.k0sproject.io/autopilot created'
    kubeletcertrotate_test.go:155: 
        	Error Trace:	/home/runner/work/k0s/k0s/inttest/kubeletcertrotate/kubeletcertrotate_test.go:155
        	            				/home/runner/work/k0s/k0s/inttest/kubeletcertrotate/kubeletcertrotate_test.go:183
        	Error:      	Received unexpected error:
        	            	the request has been made before all known HTTP paths have been installed, please try again
        	Test:       	TestKubeletCertRotateSuite/TestApply
    suite.go:77: test panicked: runtime error: invalid memory address or nil pointer dereference
        goroutine 112 [running]:
        runtime/debug.Stack()
        	/opt/hostedtoolcache/go/1.19.2/x64/src/runtime/debug/stack.go:24 +0x65
        github.com/stretchr/testify/suite.failOnPanic(0xc0007dc680, {0x23a5120, 0x3627270})
        	/home/runner/go/pkg/mod/github.com/stretchr/testify@v1.8.1/suite/suite.go:77 +0x3b
        github.com/stretchr/testify/suite.Run.func1.1()
        	/home/runner/go/pkg/mod/github.com/stretchr/testify@v1.8.1/suite/suite.go:161 +0x252
        panic({0x23a5120, 0x3627270})
        	/opt/hostedtoolcache/go/1.19.2/x64/src/runtime/panic.go:884 +0x212
        github.com/k0sproject/k0s/inttest/kubeletcertrotate.(*kubeletCertRotateSuite).applyPlan(0xc000127680, {0xc0006f6d00, 0x3})
        	/home/runner/work/k0s/k0s/inttest/kubeletcertrotate/kubeletcertrotate_test.go:156 +0x43c
        github.com/k0sproject/k0s/inttest/kubeletcertrotate.(*kubeletCertRotateSuite).TestApply(0xc000127680)
        	/home/runner/work/k0s/k0s/inttest/kubeletcertrotate/kubeletcertrotate_test.go:183 +0xc5
        github.com/k0sproject/k0s/inttest/kubeletcertrotate.(*kubeletCertRotateSuite).SetupTest(0xc000127680)
        	/home/runner/work/k0s/k0s/inttest/kubeletcertrotate/kubeletcertrotate_test.go:97 +0x7aa
        github.com/stretchr/testify/suite.Run.func1(0xc0007dc680)
        	/home/runner/go/pkg/mod/github.com/stretchr/testify@v1.8.1/suite/suite.go:165 +0x1ee
        testing.tRunner(0xc0007dc680, 0xc00056a5a0)
        	/opt/hostedtoolcache/go/1.19.2/x64/src/testing/testing.go:1446 +0x10b
        created by testing.(*T).Run
        	/opt/hostedtoolcache/go/1.19.2/x64/src/testing/testing.go:1493 +0x35f
```

### Expected behavior

The integration test should run without any panic

**Comment by twz123** [comment]
This is a recoverable error indicating that the request has been made during the startup phase of the API server and it isn't ready yet to serve the request successfully. This needs to be retired by the watcher via an appropriate error callback function.

Assigning myself.

**Comment by twz123** [comment]
(The panic is a follow-up error of the failed watch and should be fixed by #2333)
